Formula/mercurial.rb

```diff
@@ -26,6 +26,14 @@ class Mercurial < Formula
   def install
     ENV["HGPYTHON3"] = "1"
 
+    # FIXME: python@3.10 formula's "prefix scheme" patch tries to install into
+    # HOMEBREW_PREFIX/{lib,bin}, which fails due to sandbox. As workaround,
+    # manually set the installation paths to behave like prior python versions.
+    site_packages = prefix/Language::Python.site_packages("python3")
+    inreplace "Makefile",
+              "--prefix=\"$(PREFIX)\"",
+              "\\0 --install-lib=\"#{site_packages}\" --install-scripts=\"#{prefix}/bin\""
```

Comment:
Similar to #94696, need to see if this can be fixed inside `python@3.10` rather than adding workarounds to dependents.